components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -466,7 +467,7 @@ func isStsOrPodEvent(event *v1.Event) bool {
 
 func nbNameFromInvolvedObject(object *v1.ObjectReference) string {
 	nbName := object.Name
-	if object.Kind == "Pod" {
+	if object.Kind == "Pod" && strings.Contains(nbName, "-") {
```

Comment:
I'm confused about what the function nbNameFromInvolvedObject is doing

It looks to me like the code is looking for pods that are associated with the notebook. So it expects pods of the form 
${NBNAME}-0

Should we be more stringent in matching names? e.g. should we specifically look for pods named

nbname-0

So we don't match nbname-11234

Or whatever?

Can we add a unittest to catch this and other cases?

Reply:
@jlewi I think it's fine, because after that the predicate will try to find the notebook with that name and skip it if it doesn't exist.
So it's ok to be a little lax in this check.
Otherwise, I'll get to this tomorrow.

Reply:
Can we add the unittest? Since this caused a bug it would be good to have a test to prevent a regression.